File: app/src/main/java/com/dranilsaarias/nad/SICookie2.java
package com.dranilsaarias.nad;

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;
import java.net.HttpCookie;

/**
 * A simple wrapper for {@link HttpCookie} to work with {@link SiCookieStore2}
 * Gives power of serialization-deserialization to {@link HttpCookie}
 *
 * @author Manish
 */
class SICookie2 implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = 2532101328282342578L;
    private HttpCookie mHttpCookie;

    /**
     *
     */
    SICookie2(HttpCookie cookie) {
        this.mHttpCookie = cookie;
    }

    HttpCookie getCookie() {
        return mHttpCookie;
    }

    private void writeObject(ObjectOutputStream out) throws IOException {
        out.writeObject(mHttpCookie.getName());
        out.writeObject(mHttpCookie.getValue());
        out.writeObject(mHttpCookie.getComment());
        out.writeObject(mHttpCookie.getCommentURL());
        out.writeBoolean(mHttpCookie.getDiscard());
        out.writeObject(mHttpCookie.getDomain());
        out.writeLong(mHttpCookie.getMaxAge());
        out.writeObject(mHttpCookie.getPath());
        out.writeObject(mHttpCookie.getPortlist());
        out.writeBoolean(mHttpCookie.getSecure());
        out.writeInt(mHttpCookie.getVersion());
    }

    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
        String name = (String) in.readObject();
        String value = (String) in.readObject();
        mHttpCookie = new HttpCookie(name, value);
        mHttpCookie.setComment((String) in.readObject());
        mHttpCookie.setCommentURL((String) in.readObject());
        mHttpCookie.setDiscard(in.readBoolean());
        mHttpCookie.setDomain((String) in.readObject());
        mHttpCookie.setMaxAge(in.readLong());
        mHttpCookie.setPath((String) in.readObject());
        mHttpCookie.setPortlist((String) in.readObject());
        mHttpCookie.setSecure(in.readBoolean());
        mHttpCookie.setVersion(in.readInt());
    }

}

File: app/src/main/java/com/dranilsaarias/nad/SICookie2.kt
/**
 * Taken from https://github.com/loopj/android-async-http
 */
package com.dranilsaarias.nad

import java.io.IOException
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.io.Serializable
import java.net.HttpCookie

/**
 * A simple wrapper for [HttpCookie] to work with [SiCookieStore2]
 * Gives power of serialization-deserialization to [HttpCookie]
 *
 * @author Manish
 */
class SICookie2
/**
 *
 */
(cookie: HttpCookie) : Serializable {
    var cookie: HttpCookie? = null
        private set

    init {
        this.cookie = cookie
    }

    @Throws(IOException::class)
    private fun writeObject(out: ObjectOutputStream) {
        out.writeObject(cookie!!.name)
        out.writeObject(cookie!!.value)
        out.writeObject(cookie!!.comment)
        out.writeObject(cookie!!.commentURL)
        out.writeBoolean(cookie!!.discard)
        out.writeObject(cookie!!.domain)
        out.writeLong(cookie!!.maxAge)
        out.writeObject(cookie!!.path)
        out.writeObject(cookie!!.portlist)
        out.writeBoolean(cookie!!.secure)
        out.writeInt(cookie!!.version)
    }

    @Throws(IOException::class, ClassNotFoundException::class)
    private fun readObject(`in`: ObjectInputStream) {
        val name = `in`.readObject() as String
        val value = `in`.readObject() as String
        cookie = HttpCookie(name, value)
        cookie!!.comment = `in`.readObject() as String
        cookie!!.commentURL = `in`.readObject() as String
        cookie!!.discard = `in`.readBoolean()
        cookie!!.domain = `in`.readObject() as String
        cookie!!.maxAge = `in`.readLong()
        cookie!!.path = `in`.readObject() as String
        cookie!!.portlist = `in`.readObject() as String
        cookie!!.secure = `in`.readBoolean()
        cookie!!.version = `in`.readInt()
    }

    companion object {

        /**
         *
         */
        private const val serialVersionUID = 2532101328282342578L
    }

}
